src/Utility.ts

```diff
@@ -107,6 +107,32 @@ export namespace Utility {
         return port;
     }/* tslint:enable:no-any */
 
+    export async function setPort(serverXml: string, kind: Constants.PortKind, value: string): Promise<boolean> {
+        if (!await fse.pathExists(serverXml)) {
+            throw new Error(DialogMessage.noServer);
+        }
+        const xml: string = await fse.readFile(serverXml, 'utf8');
+        try {
+            /* tslint:disable:no-any */
+            const jsonObj: any = await parseXml(xml);
+            if (kind === Constants.PortKind.Server) {
+                jsonObj.Server.$.port = value;
+            } else if (kind === Constants.PortKind.Http) {
+                jsonObj.Server.Service.find((item: any) => item.$.name === Constants.CATALINA).Connector.find((item: any) =>
+                    (item.$.protocol === undefined || item.$.protocol.startsWith(Constants.HTTP))).$.port =  value;
+            } else if (kind === Constants.PortKind.Https) {
+                jsonObj.Server.Service.find((item: any) => item.$.name === Constants.CATALINA).Connector.find((item: any) =>
+                    (item.$.SSLEnabled.toLowerCase() === 'true')).$.port = value;
+            }
+            const builder: xml2js.Builder = new xml2js.Builder();
+            const newXml: string = builder.buildObject(jsonObj);
+            await fse.writeFile(serverXml, newXml);
+        } catch (err) {
+            return false;
```

Comment:
If the return type is a promise, we don't need boolean value to indicate success or failure. If the operation fails, just throw an exception and lead the caller to 'catch' block. You don't even need to catch here if you don't further process the err.